addons/web/static/src/js/fields/field_utils.js

```diff
@@ -203,24 +203,55 @@ function formatFloatFactor(value, field, options) {
  * @param {Object} [field]
  *        a description of the field (note: this parameter is ignored)
  * @param {Object} [options]
- * @param {boolean} [options.noLeadingZeroHour] if true, format like 1:30
+ * @param {string} [options.unit] gives the unit of the original float in hours, minutes or seconds.
+ *        The default is 'hours'.
+ *         Note: If the float must convert to xx:yy, no need to specify if those are minutes or hours (cfr. param below).
+ * @param {boolean} [options.threeDigits] specifies if the output format should presents the three times info => hh:mm:ss
+ *         If false (default), the float will present like xx:yy
+ *
+ *          1,52 h => 01:30:12      {'threeDigits':'True'}
+ *                 => 01:30         {}
+ *          1,52 m => 00:01:30      {'unit':'minutes', 'threeDigits':'True'}
+ *                 =>    01:30      {}
+ *        128,52 s => 00:02:09      {'unit':'seconds', 'threeDigits':'True'}
+ *                 =>    02:09      {'unit':'seconds'}
+ *
+ * @param {boolean} [options.noLeadingZero] if true, format like 1:30
```

Comment:
If this is going to be renamed, it should probably be updated to `leadingZero` defaulting to `true` instead of `noLeadingZero` defaulting to `false`. Double negatives are hard to read.

Reply:
Also the function should probably be updated to use ES6 facilities e.g. destructuring and defaults of the `options` object.

Reply:
@xmo-odoo ok for the defaults of options, but I don't get what you mean by "destructuring" sorry

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#unpacking_fields_from_objects_passed_as_a_function_parameter